themes/qgrid/source/js/menu.js

```diff
@@ -15,41 +15,193 @@ function toggleVisibility(e) {
 	}
 }
 
-function filter(e) {
+function onSearchChange(e) {
 	e.stopImmediatePropagation();
 	const input = e.target || e.srcElement;
-	const filter = input.value.toLowerCase();
+	const search = input.value.toLowerCase();
+	setSearch(search);
+}
+
+function searchOnLoad() {
+	const active = document.querySelector('.active-topic');
+	const search = getSearch();
+	if (active && search) {
+		currentPageSearch(search);
+		menuItemsSearch(search);
+	}
+}
+
+function searchOnHashChange() {
+	const search = getSearch();
+	updateMenuLinks(search);
+	currentPageSearch(search);
+	menuItemsSearch(search);
+}
+
+function currentPageSearch(search) {
+	const h2 = document.getElementsByTagName('h2');
+	const p = document.getElementsByTagName('p');
+	let scrolled = false;
+	for (let searchTarget of [...h2, ...p]) {
+		const a = searchTarget.getElementsByTagName('a')[0];
+		if (a && a.textContent) {
+			searchTarget = a;
+		}
+		if (highlightText(searchTarget, search) && !scrolled) {
+			scrolled = true;
+			searchTarget.scrollIntoView();
+		}
+	}
+}
+
+function menuItemsSearch(search) {
 	const nav = document.getElementById('nav');
-	const li = nav.getElementsByTagName('li');
-
-	for (let i = 0; i < li.length; i++) {
-		const a = li[i].getElementsByTagName('a')[0];
-		if (a) {
-			if (a.text.toLowerCase().indexOf(filter) > -1) {
-				li[i].style.display = '';
-			} else {
-				li[i].style.display = 'none';
+	const titles = nav.querySelectorAll('.title');
+	for (let title of titles) {
+		const menuItem = title.parentElement;
+		const tags = menuItem.dataset.tag || '';
+		for (let tag of tags.split(',')) {
+			if (tag.toLowerCase().indexOf(search) >= 0 && search != '') {
```

Comment:
just use like in currentPageSearch
```javascript
if(highlightText()){
  showTag
...
}
```javascript

Reply:
then `highlightText()` will be false always, because tag is empty and nothing to highlight